desc/objectives/_stability.py

```diff
@@ -367,3 +369,246 @@ def compute(self, params, constants=None):
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f.
+    bounds : tuple of {float, ndarray, callable}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+        If a callable, each should take a single argument `rho` and return the
+        desired bound (lower or upper) of the profile at those locations.
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Not used since the growth rate is always normalized.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "(normalized)"
+    _print_value_fmt = "Ideal-ballooning Stability: {:10.3e} "
+
+    def __init__(
+        self,
+        eq=None,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        deriv_mode="rev",
+        loss_function=None,
+        rho=0.5,
+        alpha=jnp.linspace(0, jnp.pi, 8)[:, None],
+        zetamax=3 * jnp.pi,
+        nzeta=200,
+        name="ideal-ball gamma",
+    ):
+        if target is None and bounds is None:
+            target = 0
+
+        self.rho = rho
+        self.alpha = alpha
+        self.zetamax = zetamax
+        self.nzeta = nzeta
+
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, eq=None, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+
+        # we need a uniform grid to get correct surface averages for iota
+        iota_grid = LinearGrid(rho=self.rho, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
```

Comment:
I'm coming around to leaving this as is rather than allowing the user to pass in a grid. Ensuring the rho values are the same is annoying and error prone, and if if we let the iota grid be user specifiable than we should also do the same for the grid used to compute `a` which starts to get messy.

That said we should probably add `sym=eq.sym` to this and the one below (double check with @unalmis on the one below).

Reply:
Yes, `a` may be computed with `sym=eq.sym`

Reply:
I thought it was easy for you to do. If it's not, let's move this custom iota grid logic to the next big PR.